smart-py/marketplace.py

```diff
@@ -1,64 +1,203 @@
+import smartpy as sp
+
 class Marketplace(sp.Contract):
     def __init__(self, objkt, metadata, manager, fee):
         self.init(
             objkt = objkt,
             metadata = metadata,
             manager = manager,
-            swaps = sp.big_map(tkey=sp.TNat, tvalue=sp.TRecord(issuer=sp.TAddress, objkt_amount=sp.TNat, objkt_id=sp.TNat, xtz_per_objkt=sp.TMutez, royalties=sp.TNat, creator=sp.TAddress)),
+            swaps = sp.big_map(
+                tkey=sp.TNat,
+                tvalue=sp.TRecord(
+                    issuer=sp.TAddress,
+                    objkt_amount=sp.TNat,
+                    objkt_id=sp.TNat,
+                    xtz_per_objkt=sp.TMutez,
+                    royalties=sp.TNat,
+                    creator=sp.TAddress
+                )
+            ),
             counter = 500000,
             fee = fee
-            )
-            
+        )
+
     @sp.entry_point
     def swap(self, params):
-        sp.verify((params.objkt_amount > 0) & ((params.royalties >= 0) & (params.royalties <= 250)))
-        self.fa2_transfer(self.data.objkt, sp.sender, sp.self_address, params.objkt_id, params.objkt_amount)
-        self.data.swaps[self.data.counter] = sp.record(issuer=sp.sender, objkt_amount=params.objkt_amount, objkt_id=params.objkt_id, xtz_per_objkt=params.xtz_per_objkt, royalties=params.royalties, creator=params.creator)
+        # previous for easy comparison
+        # sp.verify((params.objkt_amount > 0) & ((params.royalties >= 0) & (params.royalties <= 250)))
+
+        sp.verify(
+            (params.objkt_amount > 0) &
+            (params.objkt_id < self.data.counter) &
+            (params.objkt_id >= 152) &
```

Comment:
Would it make sense to declare values like these (152 and 250) as top-level constants with comments on what these values mean?

Reply:
Same as my previous comment. If the objkt_id is smaller, it will make the transfer crash, so nothing will be executed. If we do these kind of checks, they should be for user clarity and we should then add a comment in the verify explaining why the transaction failed: "The objkt_id should be higher than 151".